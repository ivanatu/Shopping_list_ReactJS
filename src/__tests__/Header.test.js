import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import renderer from 'react-test-renderer'
import Header from '../components/header';
import '../setupTests';
import {MemoryRouter} from 'react-router-dom'


describe('<Header/>', () => {
     const wrapper = shallow( <Header/> );
    
    it('renders without crashing', () => {
        shallow(<Header/>);
    });
    it('has a valid snapshot', () => { 
        const component = renderer.create(
            <MemoryRouter><Header/></MemoryRouter>); 
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders with a snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render nav without throwing an error', () => {
        expect(wrapper.exists(<nav className="nav bar"/>)).toBe(true)
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="container" />)).toBe(true)
    });
    it('should render <h1> without throwing an error', () => {
        expect(wrapper.exists(<h1 className="navbar-brand m-o align-content-end mr-auto"/>)).toBe(true)
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="navbar-nav" />)).toBe(true)
    });
    it('should render a link', () =>{
        expect(wrapper.find('Link').length).toEqual(2)
    })
});

// describe('Header.OnLogout', () => {
//     const props = {
//         history : {
//             push: {
//                 listId: 4,
//             }
//         }
//     };
//     it('returns true when called', () => {
//         const wrapper = shallow(<Header />);
//         const instance = wrapper.instance();

//         // spy on the instance instead of the component
//         spyOn(instance, 'OnLogout').and.callThrough();

//         expect(instance.OnLogout()).toBe(true);
//         expect(instance.OnLogout).toHaveBeenCalled();
//     });
// });

