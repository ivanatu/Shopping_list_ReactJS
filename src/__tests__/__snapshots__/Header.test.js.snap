// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders with a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-inverse fixed-top bg-faded navbar-toggleable-sm "
      >
        <div
          className="container"
        >
          <h1
            className="navbar-brand m-o align-content-end mr-auto"
          >
            WELCOME TO YOUR SHOPPING LIST
          </h1>
          <div
            className="navbar-nav "
          >
            <div>
              <ul>
                <div>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <i
                      className="fa fa-sign-in"
                    />
                      Login   
                  </Link>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <i
                      className="fa fa-sign-in"
                    />
                       Register  
                  </Link>
                </div>
              </ul>
              <hr />
            </div>
          </div>
        </div>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <h1
            className="navbar-brand m-o align-content-end mr-auto"
          >
            WELCOME TO YOUR SHOPPING LIST
          </h1>
          <div
            className="navbar-nav "
          >
            <div>
              <ul>
                <div>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <i
                      className="fa fa-sign-in"
                    />
                      Login   
                  </Link>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <i
                      className="fa fa-sign-in"
                    />
                       Register  
                  </Link>
                </div>
              </ul>
              <hr />
            </div>
          </div>
        </div>,
        "className": "navbar navbar-inverse fixed-top bg-faded navbar-toggleable-sm ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="navbar-brand m-o align-content-end mr-auto"
            >
              WELCOME TO YOUR SHOPPING LIST
            </h1>,
            <div
              className="navbar-nav "
            >
              <div>
                <ul>
                  <div>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                        Login   
                    </Link>
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                         Register  
                    </Link>
                  </div>
                </ul>
                <hr />
              </div>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WELCOME TO YOUR SHOPPING LIST",
              "className": "navbar-brand m-o align-content-end mr-auto",
            },
            "ref": null,
            "rendered": "WELCOME TO YOUR SHOPPING LIST",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <ul>
                  <div>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                        Login   
                    </Link>
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                         Register  
                    </Link>
                  </div>
                </ul>
                <hr />
              </div>,
              "className": "navbar-nav ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ul>
                    <div>
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                          Login   
                      </Link>
                      <Link
                        replace={false}
                        to="/register"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                           Register  
                      </Link>
                    </div>
                  </ul>,
                  <hr />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div>
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                          Login   
                      </Link>
                      <Link
                        replace={false}
                        to="/register"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                           Register  
                      </Link>
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Link
                          replace={false}
                          to="/login"
                        >
                          <i
                            className="fa fa-sign-in"
                          />
                            Login   
                        </Link>,
                        <Link
                          replace={false}
                          to="/register"
                        >
                          <i
                            className="fa fa-sign-in"
                          />
                             Register  
                        </Link>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": Array [
                            <i
                              className="fa fa-sign-in"
                            />,
                            "  Login   ",
                          ],
                          "replace": false,
                          "to": "/login",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "className": "fa fa-sign-in",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "i",
                          },
                          "  Login   ",
                        ],
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": Array [
                            <i
                              className="fa fa-sign-in"
                            />,
                            "   Register  ",
                          ],
                          "replace": false,
                          "to": "/register",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "className": "fa fa-sign-in",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "i",
                          },
                          "   Register  ",
                        ],
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  "type": "ul",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-inverse fixed-top bg-faded navbar-toggleable-sm "
        >
          <div
            className="container"
          >
            <h1
              className="navbar-brand m-o align-content-end mr-auto"
            >
              WELCOME TO YOUR SHOPPING LIST
            </h1>
            <div
              className="navbar-nav "
            >
              <div>
                <ul>
                  <div>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                        Login   
                    </Link>
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                         Register  
                    </Link>
                  </div>
                </ul>
                <hr />
              </div>
            </div>
          </div>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <h1
              className="navbar-brand m-o align-content-end mr-auto"
            >
              WELCOME TO YOUR SHOPPING LIST
            </h1>
            <div
              className="navbar-nav "
            >
              <div>
                <ul>
                  <div>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                        Login   
                    </Link>
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <i
                        className="fa fa-sign-in"
                      />
                         Register  
                    </Link>
                  </div>
                </ul>
                <hr />
              </div>
            </div>
          </div>,
          "className": "navbar navbar-inverse fixed-top bg-faded navbar-toggleable-sm ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="navbar-brand m-o align-content-end mr-auto"
              >
                WELCOME TO YOUR SHOPPING LIST
              </h1>,
              <div
                className="navbar-nav "
              >
                <div>
                  <ul>
                    <div>
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                          Login   
                      </Link>
                      <Link
                        replace={false}
                        to="/register"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                           Register  
                      </Link>
                    </div>
                  </ul>
                  <hr />
                </div>
              </div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "WELCOME TO YOUR SHOPPING LIST",
                "className": "navbar-brand m-o align-content-end mr-auto",
              },
              "ref": null,
              "rendered": "WELCOME TO YOUR SHOPPING LIST",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  <ul>
                    <div>
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                          Login   
                      </Link>
                      <Link
                        replace={false}
                        to="/register"
                      >
                        <i
                          className="fa fa-sign-in"
                        />
                           Register  
                      </Link>
                    </div>
                  </ul>
                  <hr />
                </div>,
                "className": "navbar-nav ",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ul>
                      <div>
                        <Link
                          replace={false}
                          to="/login"
                        >
                          <i
                            className="fa fa-sign-in"
                          />
                            Login   
                        </Link>
                        <Link
                          replace={false}
                          to="/register"
                        >
                          <i
                            className="fa fa-sign-in"
                          />
                             Register  
                        </Link>
                      </div>
                    </ul>,
                    <hr />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div>
                        <Link
                          replace={false}
                          to="/login"
                        >
                          <i
                            className="fa fa-sign-in"
                          />
                            Login   
                        </Link>
                        <Link
                          replace={false}
                          to="/register"
                        >
                          <i
                            className="fa fa-sign-in"
                          />
                             Register  
                        </Link>
                      </div>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <Link
                            replace={false}
                            to="/login"
                          >
                            <i
                              className="fa fa-sign-in"
                            />
                              Login   
                          </Link>,
                          <Link
                            replace={false}
                            to="/register"
                          >
                            <i
                              className="fa fa-sign-in"
                            />
                               Register  
                          </Link>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": Array [
                              <i
                                className="fa fa-sign-in"
                              />,
                              "  Login   ",
                            ],
                            "replace": false,
                            "to": "/login",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "className": "fa fa-sign-in",
                              },
                              "ref": null,
                              "rendered": null,
                              "type": "i",
                            },
                            "  Login   ",
                          ],
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": Array [
                              <i
                                className="fa fa-sign-in"
                              />,
                              "   Register  ",
                            ],
                            "replace": false,
                            "to": "/register",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "className": "fa fa-sign-in",
                              },
                              "ref": null,
                              "rendered": null,
                              "type": "i",
                            },
                            "   Register  ",
                          ],
                          "type": [Function],
                        },
                      ],
                      "type": "div",
                    },
                    "type": "ul",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
